#!/usr/bin/env python
# Motion configuration

PACKAGE = 'pandora_vision_motion'


from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

# Example : gen.add("int_param", int_t, 0, "An In.....", 50, 0, 100)
# name type level description def min max



#------------------------Background segmentation----------------------
gen.add("history", int_t, 0,"", 10, 0, 50)
gen.add("varThreshold", int_t, 0,"", 16, 4, 128)
gen.add("bShadowDetection", bool_t, 0,"", False)
gen.add("nmixtures", int_t, 0,"", 3, 1, 150)



#------------------------Threshold parameters------------------------
gen.add("diff_threshold", int_t, 0,"", 45, 0, 7500)
gen.add("motion_high_thres", double_t, 0,"", 7500, 0, 10000)
gen.add("motion_low_thres", double_t, 0,"", 200, 0, 2000)




#------------------------Visualization parameters------------------------
gen.add("visualization", bool_t, 0, "A boolean parameter for Visualization", False)
gen.add("show_image", bool_t, 0, "A boolean parameter to show current frame", False)
gen.add("show_background", bool_t, 0, "A boolean parameter to show computed background", False)
gen.add("show_diff_image", bool_t, 0, "A boolean parameter to show moving objects in current frame", False)
gen.add("show_moving_objects_contours", bool_t, 0, "A boolean parameter to show bounding rectangles around moving objects", False)

exit(gen.generate(PACKAGE, "motion_node", "motion_cfg"))
